src/components/full-view-accommodation/full-view-accommodation.tsx

```diff
@@ -21,40 +21,69 @@ export interface ShowPhotosProps {
   id: string;
   open: boolean;
   onClose: () => void;
+  handleOpen: () => void;
+  isMobile?: boolean;
 }
 
-export default function ShowPhotos({ id, open, onClose }: ShowPhotosProps) {
-  const { data } = UseGetAllMediaQuery(id);
+export default function ShowPhotos({ id, open, onClose, handleOpen, isMobile }: ShowPhotosProps) {
+  const [shouldFetch, setShouldFetch] = React.useState(false);
+  const { data } = UseGetAllMediaQuery(id, shouldFetch);
   const [currentIndex, setCurrentIndex] = React.useState(0);
 
-  const handleNext = () => {
-    setCurrentIndex((prevIndex) => Math.min(prevIndex + 1, data?.length ?? 0 - 1));
-  };
+  const handleNext = React.useCallback(() => {
+    setCurrentIndex((prevIndex) => Math.min(prevIndex + 1, (data?.length ?? 0) - 1));
+  }, [data]);
 
-  const handlePrev = () => {
+  const handlePrev = React.useCallback(() => {
     setCurrentIndex((prevIndex) => Math.max(prevIndex - 1, 0));
-  };
+  }, []);
 
   const handleClose = (event: React.MouseEvent<HTMLButtonElement>) => {
     event.stopPropagation();
     onClose();
   };
 
+  React.useEffect(() => {
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+      switch (event.key) {
+        case 'ArrowLeft':
+          handlePrev();
+          break;
+        case 'ArrowRight':
+          handleNext();
+          break;
+        default:
+          break;
+      }
+    };
+    window.addEventListener('keydown', handleKeyDown as () => void);
```

Comment:
Could you investigate the possibility of narrowing the target object from window to dialog maybe?